modules/contact/test/tests/bouncing-block-contact/frictional-mortar-fb-lm-mortar-disp.i

```diff
@@ -0,0 +1,219 @@
+    variable = disp_y
+  [../]
+[]
+
+
+[Constraints]
+  [normal_lm]
+    type = NormalMortarLMMechanicalContact
+    master_boundary = 20
+    slave_boundary = 10
+    master_subdomain = 4
+    slave_subdomain = 3
+    variable = normal_lm
+    slave_variable = disp_x
+    slave_disp_y = disp_y
+    use_displaced_mesh = true
+    compute_primal_residuals = false
+    ncp_function_type = fb
+  []
+  [normal_x]
+    type = NormalMortarMechanicalContact
+    master_boundary = 20
+    slave_boundary = 10
+    master_subdomain = 4
+    slave_subdomain = 3
+    variable = normal_lm
+    slave_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [normal_y]
+    type = NormalMortarMechanicalContact
+    master_boundary = 20
+    slave_boundary = 10
+    master_subdomain = 4
+    slave_subdomain = 3
+    variable = normal_lm
+    slave_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [tangential_lm]
+    type = TangentialMortarLMMechanicalContact
+    master_boundary = 20
+    slave_boundary = 10
+    master_subdomain = 4
+    slave_subdomain = 3
+    variable = tangential_lm
+    slave_variable = disp_x
+    slave_disp_y = disp_y
+    use_displaced_mesh = true
+    compute_primal_residuals = false
+    contact_pressure = normal_lm
+    friction_coefficient = .1
+    ncp_function_type = fb
+  []
+  [tangential_x]
+    type = TangentialMortarMechanicalContact
+    master_boundary = 20
+    slave_boundary = 10
+    master_subdomain = 4
+    slave_subdomain = 3
+    variable = tangential_lm
+    slave_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [tangential_y]
+    type = TangentialMortarMechanicalContact
+    master_boundary = 20
+    slave_boundary = 10
+    master_subdomain = 4
+    slave_subdomain = 3
+    variable = tangential_lm
+    slave_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+[]
+
+[BCs]
+  [./botx]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 40
+    value = 0.0
+  [../]
+  [./boty]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 40
+    value = 0.0
+  [../]
+  [./topy]
+    type = FunctionDirichletBC
+    variable = disp_y
+    boundary = 30
+    function = '${starting_point} * cos(2 * pi / 40 * t) + ${offset}'
+  [../]
+  [./leftx]
+    type = FunctionDirichletBC
+    variable = disp_x
+    boundary = 50
+    function = '1e-2 * t'
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 200
+  dt = 5
+  dtmin = .1
+  solve_type = 'PJFNK'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason -pc_svd_monitor -snes_linesearch_monitor -snes_ksp_ew'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount -mat_mffd_err'
+  petsc_options_value = 'lu       NONZERO               1e-15                   1e-5'
```

Comment:
Anything rather than LU will work as well?

Reply:
Yes, but the performance is quite a bit worse, perhaps because of the lagrange multiplier which introduces a saddle point. For another input file in this directory, I get 152 non-linear iterations with lu in serial, 416 non-linear iterations with ilu in serial, and 391 non-linear iterations with asm and ilu for the sub pc with two processes.

I would like to try out automatic scaling on these problems once I've introduced another option other than diagonal scaling.